src/components/utilities/validators.js

```diff
@@ -109,10 +109,24 @@ const overlappingDates = (value, allValues, meta, errorMessageKey) => {
   return undefined;
 };
 
+const amendmentWarning = (statusInAgreement, statusInLicense, statusInLicenseLabel, startDate, endDate) => {
```

Comment:
This is one of those functions that I'm afraid is gonna get tweaked and updated and messed with over and over again. People will ask that we check this field and that field, etc. And then we're gonna get stuck tacking on more and more things as function parameters.

So in cases like this, I'd prefer to just take a single object that has keys for the various things were checking, since they're all related to the amendment after all. So something like:

```
const amendmentWarning = amendment => {
  const {
    statusInAgreement,
    statusInLicense,
    startDate,
    endDate,
  } = amendment;

  if (statusInAgreement.value === statuses.CURRENT) {
    if (statusInLicense.value === statuses.EXPIRED) {
    ....
```

I went ahead and left the two `status...` entries as an object with `{ value, label }` entries because it's:
1. More flexible down the road. (Let's say we now want to use the statusInAgreement.label...)
1. Is cleaner when _calling_ this function. I know it dirties up this function a bit more, but in cases where it's six-of-one-half-dozen-of-another like this, I err on the side of cleaning up the calling experience since that'll be done more often.

Reply:
This is fair, that's a way cleaner approach. I think I agree that it's better to clean up the calling code. 

The issue is what happens when we use it in a situation where we want to use data that isn't necessarily in an amendment. In `AmendmentsFieldArray` for instance, we're calling it using `e.target` for the `statusInAgreement`, and that value obviously wouldn't be passed in with just `amendment`.

For now I can make it take an amendment and the `statusInAgreement` separately, but this feels a bit like a dirty compromise, I'd be interested to know if there was a cleaner way still.

Reply:
Yeah I'd probably mix it into the amendment itself. Eg,
```
amendmentWarning({
  ...amendment,
  statusInAgreement: e.target.value,
})
```

Reply:
Right, but if `getConflictWarnings` has code at the beginning to set `statusInAgreement` for all of those other times, then won't it always get overwritten? Unless I should be putting a check on each one to say "if this exists already, then just use that"?

Reply:
I wasn't clear, that snippet in my last message is how you'd _call_ `amendmentWarning`. So you create a new object (the amendment), spreading in all the existing amendment properties and then adding the `statusInAgreement` property.

Reply:
Ohhh I just realised you meant I could overwrite the variable, which inside the amendment is `statusForThisAgreement`, then just always have `getConflictWarnings` grab that variable

Reply:
Oh, I guess I was still off then. So I DO need `getConflictWarnings` to check whether it's being passed a `statusInAgreement`, and then if it isn't then grab `statusForThisAgreement` instead